beacon-chain/casper/validator.go

```diff
@@ -156,3 +157,56 @@ func GetProposerIndexAndShard(shardCommittees []*pb.ShardAndCommitteeArray, lcs
 	proposerIndex := slot % uint64(len(slotCommittees.ArrayShardAndCommittee[0].Committee))
 	return proposerShardID, proposerIndex, nil
 }
+
+// GetValidatorIndex returns the index of the validator given an input public key.
+func GetValidatorIndex(pubKey uint64, dynasty uint64, validators []*pb.ValidatorRecord) (uint32, error) {
+	activeValidators := ActiveValidatorIndices(validators, dynasty)
+
+	for _, index := range activeValidators {
+		if validators[index].PublicKey == pubKey {
+			return index, nil
+		}
+	}
+
+	return 0, fmt.Errorf("can't find validator index for public key %d", pubKey)
+}
+
+// GetValidatorShardID returns the shard ID of the validator currently participates in.
+func GetValidatorShardID(pubKey uint64, dynasty uint64, validators []*pb.ValidatorRecord, shardCommittees []*pb.ShardAndCommitteeArray) (uint64, error) {
```

Comment:
Prefer to not use `Get` in go function naming. Instead, just use `ValidatorShardID`. Same goes for the other functions.